apps/src/code-studio/header.js

```diff
@@ -99,83 +91,33 @@ header.build = function(
     stageExtrasEnabled
   );
 
-  /**
-   * Track boolean "visible" state of header popup to avoid
-   * expensive lookup on window resize.
-   * @type {boolean}
-   */
-  var isHeaderPopupVisible = false;
-
-  function showHeaderPopup() {
-    firehoseClient.putRecord(
-      {
-        study: 'mini_view',
-        event: 'mini_view_opened',
-        data_json: JSON.stringify({
-          current_level_id: currentLevelId
-        })
-      },
-      {includeUserId: true}
-    );
-    sizeHeaderPopupToViewport();
-    $('.header_popup').show();
-    $('.header_popup_link_glyph').html('&#x25B2;');
-    $('.header_popup_link_text').text(msg.lessAllCaps());
-    $(document).on('click', hideHeaderPopup);
-    progress.renderMiniView(
-      $('.user-stats-block')[0],
-      scriptName,
-      currentLevelId,
-      progressData.linesOfCodeText,
-      scriptData.student_detail_progress_view
+  // Hold off on rendering HeaderMiddle.  This will allow the "app load"
+  // to potentially begin before we first render HeaderMiddle, giving HeaderMiddle
+  // the opportunity to wait until the app is loaded before rendering.
```

Comment:
Oh, this is indeed for improved user experience, so that we render once after we know what we need to show, rather than rendering once and then re-rendering after app load finishes, causing UI elements to "pop" between places.  But as noted, it from my testing it appears that sometimes we need to hold off and give app load a chance to start, rather than assuming it's always started before we get to this code.